src/meetings/components/meetingAccessPoint/mediaHandlers/LocalMediaHandler.tsx

```diff
@@ -109,72 +99,70 @@ const LocalMediaHandler: FC<LocalMediaHandlerProps> = ({
 				id: `device-${i}`,
 				label: videoItem.label ? videoItem.label : `device-${i}`,
-				onClick: (): void => {
-					if (mediaDevicesEnabled.video) {
-						setEnterButtonIsEnabled(false);
-						toggleStreams(
-							mediaDevicesEnabled.audio,
-							true,
-							selectedDevicesId.audio,
-							videoItem.deviceId
-						);
-					} else {
-						setSelectedDevicesId({ audio: selectedDevicesId.audio, video: videoItem.deviceId });
-					}
-				},
-				selected: videoItem.deviceId === selectedDevicesId.video,
 				value: videoItem.deviceId
 			})),
-		[
-			mediaDevicesEnabled.audio,
-			mediaDevicesEnabled.video,
-			selectedDevicesId.audio,
-			selectedDevicesId.video,
-			setEnterButtonIsEnabled,
-			setSelectedDevicesId,
-			toggleStreams,
-			videoMediaList
-		]
+		[videoMediaList]
 	);
 
-	const mediaAudioList = useMemo(
+	const mediaAudioList: { id: string; label: string; value: string }[] = useMemo(
 		() =>
 			map(audioMediaList, (audioItem: MediaDeviceInfo, i) => ({
 				id: `device-${i}`,
 				label: audioItem.label ? audioItem.label : `device-${i}`,
-				onClick: (): void => {
-					if (mediaDevicesEnabled.audio) {
-						setEnterButtonIsEnabled(false);
-						toggleStreams(
-							true,
-							mediaDevicesEnabled.video,
-							audioItem.deviceId,
-							selectedDevicesId.video
-						);
-					} else {
-						setSelectedDevicesId({ audio: audioItem.deviceId, video: selectedDevicesId.video });
-					}
-				},
-				selected: audioItem.deviceId === selectedDevicesId.audio,
 				value: audioItem.deviceId
 			})),
+		[audioMediaList]
+	);
+
+	const onChangeVideoSource = useCallback(
+		(videoItem) => {
+			if (mediaDevicesEnabled.video) {
+				setEnterButtonIsEnabled(false);
+				toggleStreams(mediaDevicesEnabled.audio, true, selectedDevicesId.audio, videoItem);
+			}
+			setSelectedDevicesId({ audio: selectedDevicesId.audio, video: videoItem });
+		},
 		[
-			audioMediaList,
-			mediaDevicesEnabled.audio,
-			mediaDevicesEnabled.video,
+			mediaDevicesEnabled,
 			selectedDevicesId.audio,
+			setEnterButtonIsEnabled,
+			setSelectedDevicesId,
+			toggleStreams
+		]
+	);
+
+	const onChangeAudioSource = useCallback(
+		(audioItem) => {
+			if (mediaDevicesEnabled.audio) {
+				setEnterButtonIsEnabled(false);
+				toggleStreams(true, mediaDevicesEnabled.video, audioItem, selectedDevicesId.video);
+			}
+			setSelectedDevicesId({ audio: audioItem, video: selectedDevicesId.video });
+		},
+		[
+			mediaDevicesEnabled,
 			selectedDevicesId.video,
 			setEnterButtonIsEnabled,
 			setSelectedDevicesId,
 			toggleStreams
 		]
 	);
 
+	const videoSelected = useMemo(
+		() => find(mediaVideoList, ['value', selectedDevicesId.video]) ?? mediaVideoList[0],
+		[mediaVideoList, selectedDevicesId.video]
+	);
+
+	const audioSelected = useMemo(
+		() => find(mediaAudioList, ['value', selectedDevicesId.audio]) ?? mediaAudioList[0],
+		[mediaAudioList, selectedDevicesId.audio]
+	);
+
 	const updateListOfDevices = useCallback(() => {
```

Comment:
To avoid this warinng mark the type as follow

```
const audioInputs = filter(
	devices,
	(device) => device.kind === 'audioinput' && device
) as MediaDeviceInfo[];
```